api/api.go

```diff
@@ -540,6 +547,33 @@ func (api *Server) getBlockMeta(blkHash string) (*iotexapi.GetBlockMetasResponse
 	return &iotexapi.GetBlockMetasResponse{BlkMetas: []*iotextypes.BlockMeta{blockMeta}}, nil
 }
 
+// ReadState reads state on blockchain
+func (api *Server) ReadState(ctx context.Context, in *iotexapi.ReadStateRequest) (*iotexapi.ReadStateResponse, error) {
+	p, ok := api.registry.Find(string(in.ProtocolID))
+	if !ok {
+		return nil, errors.Errorf("protocol %s isn't registered", string(in.ProtocolID))
+	}
+	// TODO: need to complete the context
+	ctx = protocol.WithRunActionsCtx(ctx, protocol.RunActionsCtx{
+		BlockHeight: api.bc.TipHeight(),
+		Registry:    api.registry,
+	})
+	ws, err := api.bc.GetFactory().NewWorkingSet()
+	if err != nil {
+		return nil, err
+	}
+	data, err := util.ReadState(ctx, ws, p, in.MethodName, in.Arguments...)
+	// TODO: need to distinguish user error and system error
+	if err != nil {
+		return nil, err
+	}
+	out := iotexapi.ReadStateResponse{
+		Status: iotexapi.ReadStateResponse_SUCCESS,
```

Comment:
it's confusing that 'iotexapi.ReadStateResponse_FAILURE' is not used - there is only error return value. what is the difference?